src/ui/screens/Parachains/CrowdloanScreen.tsx

```diff
@@ -235,24 +236,11 @@ function ContributeBox({
     <Modal visible={visible} onDismiss={reset}>
       <Text>Contribute with:</Text>
       <Padder scale={0.5} />
-      <SelectAccount onSelect={(selectedAccount) => setAccount(selectedAccount.address)} />
+      <SelectAccount onSelect={(selectedAccount) => setAccount(selectedAccount.accountInfo)} />
       <Padder scale={1} />
       <Text>Amount:</Text>
-      <TextInput
-        style={contributeBoxStyles.textInput}
-        mode="outlined"
-        autoComplete="off"
-        placeholder="Enter amount"
-        keyboardType="decimal-pad"
-        value={amount}
-        onFocus={() => setAmount('')}
-        onChangeText={(nextValue) => setAmount(decimalKeypad(nextValue))}
-        contextMenuHidden={true}
-        right={<TextInput.Affix text={(api && formatBalance(getBalanceFromString(api, amount))) ?? ''} />}
-      />
-      <MaxBalance address={account} />
+      {api && account && <BalanceInput api={api} account={account} onChangeBalance={setAmount} />}
```

Comment:
Same here. The `BalanceInput` component wont render until there is a selected account.